eyeball-im-util/src/vector.rs

```diff
@@ -2,4 +2,96 @@
 
 mod filter;
 
+use eyeball_im::VectorDiff;
+use futures_core::Stream;
+
 pub use self::filter::{Filter, FilterMap};
+
+/// Abstraction over stream items that the adapters in this module can deal
+/// with.
+pub trait VectorDiffContainer: VectorDiffContainerOps<Self::Element> {
```

Comment:
These traits are yummy. Well done!

Reply:
Thanks! Took quite some time to get something that works ^^